### PR TITLE
code to hide done button above keyboard for iOS

### PR DESCRIPTION
**Pull request by vikassharma96**
## Summary

While using `keyboardType` (all number pads) with TextInput and `returnKeyType="done"` it shows a default Done button above the keyboard. To make it configurable for iOS I added the `hideDoneButtonIOS` option. It will allow to show/hide the default Done button when we're using number pads as keyboardType with TextInput.

## Changelog

[iOS] [Added] - Added hideDoneButtonIOS as an optional prop to TextInput

## Test Plan

Functionally wise there shouldn't be much to break the API. We'd pass it like others props we're passing to TextInput.
```
<TextInput hideDoneButtonIOS={true} />
```

### CODE REVIEW

File: Libraries/Components/TextInput/TextInput.js [hunk 1]
```diff
@@ -301,6 +301,13 @@ type IOSProps = $ReadOnly<{|
    * @platform ios
    */
   textContentType?: ?TextContentType,
+
+  /**
+   * If `true`, it will hide the done button above the keyboard
+   * Only work for any number pads keyboardType
+   * @platform ios
+   */
+   hideDoneButtonIOS?: ?boolean,
```
Comment (dmitryrykun) [inline Libraries/Components/TextInput/TextInput.js:310]: 1. Please rename it to `showDoneButtonForNumberPadKeyboard`.
2. Remove the negation it the place where this value is used.
3. Update the above description.
4. Delete extra whitespace before this property name.

Comment (vikassharma96) [reply Libraries/Components/TextInput/TextInput.js:310]: @dmitryrykun thank you for reviewing this PR. I have done the changes as mentioned but couldn’t figure it out to set the default value of showDoneButtonForNumberPadKeyboard to true.

Comment (dmitryrykun) [reply Libraries/Components/TextInput/TextInput.js:310]: @vikassharma96  that's great, `initWithBridge` is usually the best place to initialise object properties. Please push your changes.

**Comment by vasylnahuliak** [comment]
Up

**Pull request closed without merging**